src/components/vuestic-components/va-form/VaForm.vue

```diff
@@ -0,0 +1,162 @@
+<script>
+import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+
+const availableFormTags = ['VaInput', 'VaColorInput', 'VaDatePicker', 'VaCheckbox', 'VaRadio', 'VaSelect', 'VaFileUpload']
+const getAllChildren = (vm, children = []) => {
+  vm.$children.forEach(function (child) {
+    children.push(child)
+    child.$children.length > 0 && getAllChildren(child, children)
+  })
+  return children
+}
+
+export default {
+  name: 'VaForm',
+  mixins: [ContextPluginMixin],
+  props: {
+    lazyValidation: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'lazyValidation', false)
+      },
+    },
+    autofocus: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'autofocus', false)
+      },
+    },
+    tag: {
+      type: String,
+      default () {
+        return getContextPropValue(this, 'tag', 'div')
+      },
+    },
+  },
+  data () {
+    return {
+      valid: true,
+      hasResetButton: false,
```

Comment:
Same here. Buttons can be added dynamically and state won't play nicely.

Reply:
These fields need to watch native form event like a submit and reset in cases when a form has tag div. Probably we won't use dynamically buttons with type submit and reset

Reply:
Couple of approaches for reference.

* `this.$form.reset()`, `this.$form.submit()`. `$form` here could be inject/provided as per usual. Probably the top level form would make more sense to have priority in case we have nested forms.
* Let the user handle it on its own. So to get reset - update model, to submit the form - do `this.$refs.form.submit()`.